### PR TITLE
Let new-storage DynamoDb use Apache Http Client

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: versioned/storage/dynamodb/src/main/java/org/projectnessie/versioned/storage/dynamodb/DynamoClientProducer.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
 import java.net.URI;
 import org.immutables.value.Value;
 import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
-import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
```
Comment (XN137) [inline versioned/storage/dynamodb/src/main/java/org/projectnessie/versioned/storage/dynamodb/DynamoClientProducer.java:21]: whats the reasoning for only doing this change here but not also in `DynamoDatabaseClient.java` ?

File: versioned/storage/dynamodb/build.gradle.kts [hunk 2]
```diff
@@ -44,11 +44,8 @@ dependencies {
   implementation(libs.slf4j.api)
 
   implementation(platform(libs.awssdk.bom))
-  implementation("software.amazon.awssdk:dynamodb") {
-    exclude("software.amazon.awssdk", "apache-client")
-  }
-  implementation("software.amazon.awssdk:netty-nio-client")
-  implementation("software.amazon.awssdk:url-connection-client")
```
Comment (XN137) [inline versioned/storage/dynamodb/build.gradle.kts:51]: we seem to be using `software.amazon.awssdk:url-connection-client` in lots of other places as well... wouldnt it make sense to switch all awssdk usage to the same apache http client?

**Review by XN137 — APPROVED** [review]
change looks right for new dynamodb storage but not sure i understand why other parts are not switched over as well

File: versioned/storage/dynamodb/src/main/java/org/projectnessie/versioned/storage/dynamodb/DynamoClientProducer.java (hunk 1 above)
Comment (snazy) [reply versioned/storage/dynamodb/src/main/java/org/projectnessie/versioned/storage/dynamodb/DynamoClientProducer.java:21]: That's code that'll go away.

File: versioned/storage/dynamodb/build.gradle.kts (hunk 2 above)
Comment (snazy) [reply versioned/storage/dynamodb/build.gradle.kts:51]: Others are testing modules

Comment (XN137) [reply versioned/storage/dynamodb/build.gradle.kts:51]: if the apache client is the preferred choice for production modules, why not also for testing modules?

**Pull request merged**